clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -360,7 +360,7 @@ public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
                         .setPartitions(partitionResponses));
             });
         }
-        return new FetchResponse(new FetchResponseData()
+        return FetchResponse.of(new FetchResponseData()
```

Comment:
For versions prior to 13, the data is iterated twice. To prevent this redundancy, consider making the `FetchResponse` constructor package-private to avoid the double loop.

Reply:
Make sense.